### PR TITLE
Feature/filter farms

### PR DESCRIPTION
**Pull request by heyimbrucelee**
feat: Add a filter input on Pools & Farms
Removed FarmTabButtons & PoolTabButtons
Created new component TabToolbar
stakeOnly button : now filter inactive lists too
stakeOnly button : only displayed when user logged in
useI18n : fix

### CODE REVIEW

File: src/components/TabToolbar/TabToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  setFilter: (boolean) => void
+}
+
+const TabToolbar: React.FC<TabToolbarProps> = ({ account, stakedOnly, setStakedOnly, filter, setFilter }) => {
+  const { url, isExact } = useRouteMatch()
+  const TranslateString = useI18n()
+
+  return (
+    <Wrapper>
+      <>
```
Comment (RabbitDoge) [inline src/components/TabToolbar/TabToolbar.tsx:21]: unnecessary fragment

File: src/components/TabToolbar/TabToolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        value={filter}
+        onChange={(e) => setFilter(e.target.value)}
+      />
+    </Wrapper>
+  )
+}
+
+export default TabToolbar
+
+const Wrapper = styled.div`
```
Comment (RabbitDoge) [inline src/components/TabToolbar/TabToolbar.tsx:50]: set the styled components at the top

File: src/hooks/useI18n.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ const useI18n = () => {
 
   return (translationId: number, fallback: string, data: ContextData = {}) => {
     const foundTranslation = translations.find((translation) => {
-      return translation.data.stringId === translationId
+      return translation.data?.stringId === translationId
```
Comment (RabbitDoge) [inline src/hooks/useI18n.ts:14]: Can you explain why this change ?

**Review by RabbitDoge — CHANGES_REQUESTED** [review]
conflict needs to be solved

File: src/hooks/useI18n.ts (hunk 3 above)
Comment (heyimbrucelee) [reply src/hooks/useI18n.ts:14]: avoid app crash if translations cant get load

**Comment by heyimbrucelee** [comment]
> conflict needs to be solved

i merged with origin/develop and had no conflict to solve.

Conflicting files
src/views/Farms/components/FarmTabButtons/index.tsx
is deleted

**Comment by hachiojidev** [comment]
@heyimbrucelee Looks like there is a conflict in FarmTabButton. Don't forget to pull the latest develop. Personally I recommend rebasing
image

**Comment by JayJayFinance** [comment]
> conflict needs to be solved

I will be finished soon, thank

**Comment by heyimbrucelee** [comment]
@Chef-Chungus hi Chef ty for understanding my point of view. i appreciate it.

Plz consider merging this feature as it could dramatically improve user experience waiting for a more advanced card/list view.
it would be a pleasure to continue contributing to the project at my level !
Regards

**Comment by RabbitDoge** [comment]
@heyimbrucelee If we don't have any news from you soon, I'll close this pr

**Comment by heyimbrucelee** [comment]
> @heyimbrucelee If we don't have any news from you soon, I'll close this pr

Haha, I wish i could give news and contribute more but i not welcome since 'someone' fired me from slack and this pr has been ignored for 10+ days.

Anyway there is filter in farms now (cool), and the pools will probably get same rework.
But it could have been useful meanwhile.

**Comment by BaptisteGarcin** [comment]
Why did they kick you? @heyimbrucelee It looks like you took a lot of your time to contribute and that your PR was ok at the time

**Comment by Chef-Chungus** [comment]
> Why did they kick you? @heyimbrucelee It looks like you took a lot of your time to contribute and that your PR was ok at the time @BaptisteGarcin 

Bruce wasn't kicked for this PR, he was previously kicked for complaining about compensation not being paid in a timely enough fashion for a task that initially wasn't issued as a bounty, and he did in his own regard. 
We are grateful that he wishes to help the community and PCS by contributing to the development, and we were willing to reward him for this, however, he was unappreciative of our efforts to do this as the payment wasn't paid fast enough. 


Due to the latest farm updates we've done, we don't believe this PR is relevant anymore and will be closing it. 

Thank you for taking the time to submit this, @heyimbrucelee. We appreciate your passion for the project and community.

**Comment by heyimbrucelee** [comment]
> > Why did they kick you? @heyimbrucelee It looks like you took a lot of your time to contribute and that your PR was ok at the time @BaptisteGarcin
> 
> Bruce wasn't kicked for this PR, he was previously kicked for complaining about compensation not being paid in a timely enough fashion for a task that initially wasn't issued as a bounty, and he did in his own regard.
> We are grateful that he wishes to help the community and PCS by contributing to the development, and we were willing to reward him for this, however, he was unappreciative of our efforts to do this as the payment wasn't paid fast enough.
> 
> Due to the latest farm updates we've done, we don't believe this PR is relevant anymore and will be closing it.
> 
> Thank you for taking the time to submit this, @heyimbrucelee. We appreciate your passion for the project and community.


My first PR was for adding toggle 'staked only' button on farms/pools.
Then you invited me, ask me to make code modifications etc.. and told me we want to reward you 200$ in $cake send me your wallet address. (8 feb)
https://github.com/pancakeswap/pancake-frontend/pull/416

Second was adding $values under $cake values on home then you told me same we want to reward you 200$ in $cake send me your wallet address. (12 feb)
https://github.com/pancakeswap/pancake-frontend/pull/447 

I never asked compensation you offered it, i waited to get paid quietly
I remember i tried to ask you during the berry IFO (16 feb) (but now we know IFOs are fonky @ PCS  XD)
Waited another day...
Then on 18/19 feb i politely asked if i finally could get paid and if you could take in consideration that when PR was merged and you told me about reward / ask my walled address $cake ~6/7$.

nothing about "complaining about compensation not being paid in a timely enough fashion"
i only asked you to be fair since $cake isnt a stablecoin and (200$ cake @6/7) ==! (200$ cake@15)

after waiting 5/6/7 days and watching cake  :rocket:  :moon:  
the only day i really asked you to honor your words you kicked me from slack
i had really bad night with strange nightmares rabbits trying to kill me with pancakes dipping me in syrup and so on (ok this part is not true actually)
then in the morning i woke up with 60 cakes (400$ cake@6.6)
so you have been fair and i thanked you.
but honestly cant understand why you decided to kick me. easier/faster than debating/sharing pov for sure.

after this, i commented on this PR wich was ready to merge so i asked to consider merging it cause i was a good temp solution imo.

Due to the latest farm updates we've done, we don't believe this PR is relevant anymore and will be closing it.
--> sure it was 18days ago XD

Thank you for taking the time to submit this, @heyimbrucelee. We appreciate your passion for the project and community.
--> as i said feel free to invite me back

be water my friends XD
